#! /bin/sh /usr/share/dpatch/dpatch-run
## 00_0177-qurl-clear-fix.dpatch by Trolltech ASA
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix QUrl::clear().

@DPATCH@
diff -urNad qt4-x11-4.3.0~/src/corelib/io/qurl.cpp qt4-x11-4.3.0/src/corelib/io/qurl.cpp
--- qt4-x11-4.3.0~/src/corelib/io/qurl.cpp	2007-05-25 15:24:10.000000000 +0200
+++ qt4-x11-4.3.0/src/corelib/io/qurl.cpp	2007-06-30 19:52:57.000000000 +0200
@@ -3421,6 +3421,7 @@
     fragment.clear();
 
     encodedOriginal.clear();
+    encodedNormalized.clear();
 
     isValid = false;
     hasQuery = false;
@@ -3429,7 +3430,7 @@
     valueDelimiter = '=';
     pairDelimiter = '&';
 
-    QURL_UNSETFLAG(stateFlags, Parsed | Validated);
+    QURL_UNSETFLAG(stateFlags, Parsed | Validated | Normalized);
 }
 
 QByteArray QUrlPrivate::toEncoded(QUrl::FormattingOptions options) const
